### ISSUE TITLE
sender crashes when receiver enters wrong code

### ISSUE THREAD

**Issue by yojoe** [Jacalz/wormhole-gui#6]
When the receiver enters a wrong wormhole code, the sender (wormhole-gui) crashes with the following output in the terminal:

```
Fyne error:  Error on sending text
At: /app/internal/bridge/widgets/list_send.go:158
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x8bae36]

goroutine 45 [running]:
fyne.io/fyne/dialog.NewError(0x0, 0x0, 0xce3ea0, 0xc0000a91e0, 0x0, 0x1514840)
  /go/pkg/mod/fyne.io/fyne@v1.4.2/dialog/information.go:37 +0x26
fyne.io/fyne/dialog.ShowError(0x0, 0x0, 0xce3ea0, 0xc0000a91e0)
  /go/pkg/mod/fyne.io/fyne@v1.4.2/dialog/information.go:43 +0x49
github.com/Jacalz/wormhole-gui/internal/bridge/widgets.(*SendList).SendText.func1(0xc000081a40, 0x0, 0x0, 0xc000212270, 0xc000081a40)
  /app/internal/bridge/widgets/list_send.go:159 +0x1b8
created by github.com/Jacalz/wormhole-gui/internal/bridge/widgets.(*SendList).SendText
  /app/internal/bridge/widgets/list_send.go:156 +0x3a5
```

**Comment by Jacalz** [comment]
Nice catch! Thanks for opening this issue. It looks like this is an issue with the upstream Fyne toolkit when the error dialog receives a `nil` error. Will make sure to get that fixed :+1:

**Comment by Jacalz** [comment]
There also seems to be a second issue where the message could be nil when trying to reject the transfer.

**Comment by Jacalz** [comment]
This is now fixed in the `develop` branch and will be available in the 2.1.0 release at the end of January or early February.

**Comment by Jacalz** [comment]
I changed my mind and did a quick 2.0.2 update with this change included. Downloading from https://github.com/Jacalz/wormhole-gui/releases/tag/v2.0.2 should fix your issue 🙂
